test/SlowTests/Issues/RavenDB-10343.cs

```diff
@@ -0,0 +1,1375 @@
+                    var query = session.Query<Build>()
+                        .Where(entry => entry.ProductKey.In(productBuildKeys) && entry.IsPublic)
+                        .Where(entry => entry.Channel.In(buildTypes))
+                        .Select(entry => session.Load<TeamCityBuild>(entry.TeamCityBuildLocalId))
+                        .Select(build => build.BuildDate);
+
+                    Assert.Equal("from Builds as entry " +
+                                 "where (entry.ProductKey in ($p0) and entry.IsPublic = $p1) " +
+                                 "and (entry.Channel in ($p2)) " +
+                                 "load entry.TeamCityBuildLocalId as build " +
+                                 "select build.BuildDate"
+                        , query.ToString());
+
+                    var queryResult = query.ToList();
+
+                    Assert.Equal(2, queryResult.Count);
+
+                    Assert.Equal(new DateTime(2016, 6, 6), queryResult[0]);
+                    Assert.Equal(new DateTime(2018, 1, 1), queryResult[1]);
+
+                }
+            }
+        }
+
+        [Fact]
+        public void CanQueryWithLoadFromSelectWithoutProjection()
+        {
+            using (var store = GetDocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Build
+                    {
+                        ProductKey = "products/1-A",
+                        IsPublic = true,
+                        Channel = "types/1",
+                        TeamCityBuildLocalId = "teamCityBuilds/1-A"
+
+                    });
+
+                    session.Store(new Build
+                    {
+                        ProductKey = "products/2-A",
+                        IsPublic = true,
+                        Channel = "types/2",
+                        TeamCityBuildLocalId = "teamCityBuilds/2-A"
+
+                    });
+
+                    session.Store(new TeamCityBuild
+                    {
+                        DownloadsIds = new List<string>
+                        {
+                            "buildDownloads/1-A",
+                            "buildDownloads/2-A"
+                        }
+                    }, "teamCityBuilds/1-A");
+
+                    session.Store(new TeamCityBuild
+                    {
+                        DownloadsIds = new List<string>
+                        {
+                            "buildDownloads/1-A",
+                            "buildDownloads/3-A",
+                            "buildDownloads/4-A"
+                        }
+                    }, "teamCityBuilds/2-A");
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Query<Build>()
+                        .Select(entry => RavenQuery.Load<TeamCityBuild>(entry.TeamCityBuildLocalId));
```

Comment:
I wanted to check how it works if we select a property of a loaded document. I added `.BuildDate` :

`.Select(entry => RavenQuery.Load<TeamCityBuild>(entry.TeamCityBuildLocalId).BuildDate);`

and it threw:

```
System.ArgumentException : Could not understand expression: from Builds.Select(entry => Load(entry.TeamCityBuildLocalId).BuildDate)
---- System.ArgumentException : Not supported computation: Load(entry.TeamCityBuildLocalId).BuildDate. You cannot use computation in RavenDB queries (only simple member expressions are allowed).
```

Reply:
fixed